version: "3.8"

services:
  cadvisor:
    image: gcr.io/cadvisor/cadvisor:v0.46.0
    volumes:
      - /:/rootfs:ro
      - /var/run:/var/run:ro
      - /sys:/sys:ro
      - /var/lib/docker/:/var/lib/docker:ro
    deploy:
      mode: global
      restart_policy:
        condition: on-failure
      labels:
        - "traefik.enable=true"
        - "traefik.docker.network=traefik"
        - "traefik.http.routers.cadvisor.rule=Host(`cadvisor.localhost`)"
        - "traefik.http.routers.cadvisor.entrypoints=web"
        - "traefik.http.routers.cadvisor.service=cadvisor"
        - "traefik.http.services.cadvisor.loadbalancer.server.port=8080"
    networks:
      - monitoring
      - traefik

networks:
  monitoring:
    external: true
  traefik:
    external: true